rules/ast/ast.go

```diff
@@ -376,6 +376,7 @@ func (node *VectorAggregation) labelsToGroupingKey(labels clientmodel.Metric) ui
 	summer := fnv.New64a()
 	for _, label := range node.groupBy {
 		fmt.Fprint(summer, labels[label])
+		fmt.Fprint(summer, []byte{0})
```

Comment:
Done. I also noticed that the method below this had the same bug, but it could simply be replaced by a whole metric fingerprint. I removed that method and included a test for that too.